### PR TITLE
Improve unbacked span

### PR DESCRIPTION
**Pull request by bkchr**
We need to make sure to drop the import-statement child span before the
parent span is dropped.

(There are still ways the parent is dropped before, however I think these are negligible, as this only happens in the error case.)

### CODE REVIEW

File: node/core/backing/src/lib.rs [hunk 1]
```diff
@@ -680,31 +680,41 @@ impl CandidateBackingJob {
...
+				if let Some(backed) =
+					table_attested_to_backed(attested, &self.table_context)
+				{
+					let message = ProvisionerMessage::ProvisionableData(
+						self.parent,
+						ProvisionableData::BackedCandidate(backed.receipt()),
+					);
+					self.send_to_provisioner(message).await?;
+
+					span.as_ref().map(|s| s.child("backed"));
```
Comment (ordian) [inline node/core/backing/src/lib.rs:700]: are we creating and dropping a child span here? this warrants some comments

**Review by coriolinus — APPROVED** [review]
Seems like a straightforward fix for this case, so I'm approving, but this feels like something we could generalize.

We could adjust `JaegerSpan` such that each child contains a reference (or `Rc` or something) to its parent. This would ensure that we _never_ drop a child before its parent, by virtue of the type system. I think encoding domain logic into the type system like that is desirable.

File: node/core/backing/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply node/core/backing/src/lib.rs:700]: Why does this warrants a comment? 

It should just add a span says "backed" to see in the UI when a unbacked candidate was backed.

Comment (ordian) [reply node/core/backing/src/lib.rs:700]: it's not obvious that we encode a state transition with an ephemeral child span that is dropped immediately, I'm not sure this is the intended use of spans

Comment (bkchr) [reply node/core/backing/src/lib.rs:700]: The whole "unbacked" story is probably not the intended use of spans. :shrug:

**Pull request merged**